src/Microsoft.OData.Core/JsonLight/JsonLightMetadataLevel.cs

```diff
@@ -24,10 +24,11 @@ internal abstract class JsonLightMetadataLevel
         /// </summary>
         /// <param name="mediaType">The full media type being written. This media type must have a type/subtype of "application/json".</param>
         /// <param name="metadataDocumentUri">The metadata document uri from the writer settings.</param>
+        /// <param name="alwaysAddTypeAnnotationsForDerivedTypes">When set, type annotations will be added for derived types, even when the metadata level is set to "None".</param>
         /// <param name="model">The edm model.</param>
         /// <param name="writingResponse">true if we are writing a response, false otherwise.</param>
         /// <returns>The JSON Light metadata level being written.</returns>
-        internal static JsonLightMetadataLevel Create(ODataMediaType mediaType, Uri metadataDocumentUri, IEdmModel model, bool writingResponse)
+        internal static JsonLightMetadataLevel Create(ODataMediaType mediaType, Uri metadataDocumentUri, bool alwaysAddTypeAnnotationsForDerivedTypes, IEdmModel model, bool writingResponse)

```

Comment:
Maybe add an "Create" overload without alwaysAddTypeAnnotationsForDerivedTypes parameter and call "Create" with alwaysAddTypeAnnotationsForDerivedTypes using "false".

Reply:
`Create` overload added in the 3rd commit to this PR.